TcgPocket/Features/Users/UsersController.cs

```diff
@@ -0,0 +1,106 @@
+
+namespace TcgPocket.Features.Users;
+
+[ApiController]
+[Route("/users")]
+public class UsersController : ControllerBase
+{
+    private readonly IMediator _mediator;
+
+    public UsersController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<Response<List<UserDto>>>> GetAllUsers()
+    {
+        var response = await _mediator.Send(new GetAllUsersQuery());
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
+
+    [HttpGet("{id:int}", Name = nameof(GetUserById))]
+    public async Task<ActionResult<Response<UserGetDto>>> GetUserById([FromRoute] int id)
+    {
+        var response = await _mediator.Send(new GetUserByIdQuery { Id = id });
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+
+    [HttpGet("{id:int}/roles", Name = nameof(GetAllRolesByUserId))]
+    public async Task<ActionResult<Response<List<RoleGetDto>>>> GetAllRolesByUserId([FromRoute] int id)
+    {
+        var response = await _mediator.Send(new GetAllRolesByUserIdQuery { UserId = id });
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<Response<UserGetDto>>> CreateUser([FromBody] UserCreateDto data)
+    {
+        var response = await _mediator.Send(new CreateUserCommand { User = data });
+
+        return response.HasErrors
+            ? BadRequest(response)
+            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
+    }
+
+    [HttpPost("{id:int}/roles")]
+    public async Task<ActionResult<Response<UserRoleDto>>> AddRoleToUser([FromRoute] int id, [FromBody] AddRoleToUserCommand data)
+    {
+        var response = await _mediator.Send(new AddRoleToUserCommand { Id = id, RoleName = data.RoleName });
+
+        return response.HasErrors
+            ? BadRequest(response)
+            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
+    }
+
+    [HttpPut("{id:int}")]
+    public async Task<ActionResult<Response<UserGetDto>>> UpdateUser([FromRoute] int id,
+        [FromBody] UserDto data)
+    {
+        var response = await _mediator.Send(new UpdateUserCommand { Id = id, User = data });
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
+
+    [HttpPut("{id:int}/password-update", Name = nameof(UpdatePassword))]
+    public async Task<ActionResult<Response<UserGetDto>>> UpdatePassword(
+        [FromBody] UpdatePasswordCommand data)
+    {
+        var response = await _mediator
+            .Send(new UpdatePasswordCommand 
+            { 
+                UserName = data.UserName, 
+                CurrentPassword = data.CurrentPassword, 
+                NewPassword = data.NewPassword, 
+                NewPasswordConfirmation = data.NewPasswordConfirmation 
+            });
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
```

Comment:
If we are going to get the whole command from the body, we might as well just use it 😄 
Also, we don't use the `id` here so it can be removed from the route.

```suggestion
    [HttpPut("password-update", Name = nameof(UpdatePassword))]
    public async Task<ActionResult<Response<UserGetDto>>> UpdatePassword(
        [FromBody] UpdatePasswordCommand data)
    {
        var response = await _mediator.Send(data);

        return response.HasErrors ? BadRequest(response) : Ok(response);
    }
```